### DISCUSSION TITLE
What should go into an eventual 1.2.3 release?

### DISCUSSION THREAD

**Discussion by aesedepece** [witnet/witnet-rust#1964] [General]
There's this intention of releasing `1.2.3` asap. This is to discuss what should we ship with it, and what can be left out.

**Comment by aesedepece** [comment]
The https://github.com/witnet/witnet-rust/tree/202106-recovery branch introduced 11 commits that are not on `upstream/master`.

We may need to backport some of those changes to the master branch and make sure that goes into an eventual `1.2.3` release.

Shall we make a list of what is worth backporting?

**Reply by aesedepece** [reply]
Namely, these were the changes:
https://github.com/witnet/witnet-rust/compare/202106-recovery

**Reply by aesedepece** [reply]
At first thought, commit c97bd5fbed83a35c6ebdc85277fac1db4b7ceddd should be left out, as these pieces of code are no longer needed:

https://github.com/witnet/witnet-rust/blob/d8cf21857d8466b783af640c992af2e01a12c8e8/data_structures/src/mainnet_validations.rs#L14-L22

https://github.com/witnet/witnet-rust/blob/d8cf21857d8466b783af640c992af2e01a12c8e8/data_structures/src/mainnet_validations.rs#L44-L53

Which makes me think that the code for the first emergency committee can be removed just as well, right?

**Reply by drcpu-github** [reply]
I am wondering if we want to remove this. Does it not make sense to keep it included for posterity, so in the future people can easily verify things like emergency committees and how they worked? I cannot immediately think of a use case for that, but I'm also not sure if removing them serves a good purpose.

**Reply by aesedepece** [reply]
That was also the reason for keeping the logic of the first emergency committee in place.

My point with removing any piece of dead code (those parts that are no longer used for validations, mining nor synchronization) is to maintain our code base as lean and maintainable as possible.

In the same fashion, my take is that once some protocol upgrade has been enacted via TAPI, the following `witnet-rust` release should remove the conditional activation from that part of the code, because the logic contained therein will have become part of the core protocol as per [WIP-0014§17](https://github.com/witnet/WIPs/blob/master/wip-0014.md#eventual-activation-of-a-protocol-improvement-that-is-supported-enough).

**Reply by drcpu-github** [reply]
> My point with removing any piece of dead code (those parts that are no longer used for validations, mining nor synchronization) is to maintain our code base as lean and maintainable as possible.
> 

I'd say the fact that it is put in a separate function makes it pretty lean / maintainable. I have no strong opinion on keeping or removing it, because I cannot

> In the same fashion, my take is that once some protocol upgrade has been enacted via TAPI, the following `witnet-rust` release should remove the conditional activation from that part of the code, because the logic contained therein will have become part of the core protocol as per [WIP-0014§17](https://github.com/witnet/WIPs/blob/master/wip-0014.md#eventual-activation-of-a-protocol-improvement-that-is-supported-enough).

I feel like this is a slightly different case because those are protocol improvements versus interventions in the consensus mechanism, but in the end the repository and associated tags show a good enough overview of the different changes the node software went through.

**Reply by aesedepece** [reply]
> I'd say the fact that it is put in a separate function makes it pretty lean / maintainable. I have no strong opinion on keeping or removing it, because I cannot

I dislike the idea of having `if` / `else if` / `else` blocks covering conditions that we know that can never be met, because each of those conditions add cognitive complexity to the business logic and also require a specific test.

**Comment by drcpu-github** [comment]
I guess all commits, except 845e3362 (shorter icing period) and 6ed2de93 (microforking) make sense to include in the master branch.
